### ISSUE TITLE
Crash observed during load test when Sending response to an incoming http request from a different thread

### ISSUE THREAD

**Issue by zanpakuto** [ipkn/crow#277]
**Steps:**

- The lambda which is specified under the rule is invoked by crow when a request is received.
- A new thread is spawned from the listener to handle the request.
- response is triggered from the new thread
- Crash is observed after few seconds

**Sample code:**
`
void mockhandler(const crow::request & req, crow::response & res) {
    res->end();
}

CROW_ROUTE(app, "/hello").methods("POST" __method) {
    [](const crow::request & req, crow::response & res) {
        std::thread(mockhandler, &req, &res).detach();
    }
}`

However, the crash is NOT observed when the response is sent from the lambda itself as below.
`CROW_ROUTE(app, "/hello").methods("POST" __method) {
    [](const crow::request & req, crow::response & res) {
        return crow::response(200);
    }
}`

**Comment by jjhoo** [comment]
My guess is that `res` in `mockhandler` may point to an object that has been deallocated since `res` in that route-method is essentially just a temporary object. I doubt there is any support for this kind of delayed execution.

I would try the following if you are trying to handle a situation where preparing the response takes a long time:
 1) start a work thread that updates a status object periodically
 2) return '202' response with a temporary location
 3) return some status information from that temporary location until the work has been finished
 4) return '303' response from that temporary location when the work has been finished

based on http://restcookbook.com/Resources/asynchroneous-operations/
